workers/unity/Packages/com.improbable.gdk.tools/.CodeGenerator/CodeGeneration/Utils/Formatting.cs

```diff
@@ -0,0 +1,59 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+namespace Improbable.Gdk.CodeGeneration.Utils
+{
+    public static class Formatting
+    {
+        public static string CapitaliseQualifiedNameParts(string text)
+        {
+            var parts = text.Split(new[] { "." }, StringSplitOptions.RemoveEmptyEntries)
+                .Select(s =>
+                    SnakeCaseToPascalCase(char.ToUpperInvariant(s[0]) + s.Substring(1, s.Length - 1)))
+                .ToArray();
+            return string.Join(".", parts);
+        }
+
+        public static string SnakeCaseToPascalCase(string text)
+        {
+            return ToPascalCase(text.Split(new[] { "_" }, StringSplitOptions.RemoveEmptyEntries));
+        }
+
+        private static string ToPascalCase(IEnumerable<string> parts)
```

Comment:
Does this need to be separate? It is only used in SnakeCaseToPascalCase which is quite simple on itself

Reply:
Will merge them 👍